onnxruntime/core/framework/tensor.cc

```diff
@@ -47,6 +47,21 @@ size_t Tensor::SizeInBytes() const {
   return ret;
 }
 
+Tensor Tensor::Clone(AllocatorPtr allocator) const {
+  Tensor cloned(dtype_, shape_, allocator, byte_offset_);
```

Comment:
Not sure if there is a device specific guard required. This logic is obviously only for Tensors whose data is in CPU. Need to follow-up, and hence keeping this as a placeholder comment.